### PR TITLE
refatored conferences_controller

### PR DESCRIPTION
**Pull request by carlosdanielpohlod**
**Checklist**

- [ x ] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x  ] My branch is up-to-date with the upstream `master` branch.
- [x  ] The tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [ ] I have added necessary documentation (if appropriate).

**Short description of what this resolves/which [issues](https://github.com/openSUSE/osem/issues) does this fix?:**

-

**Changes proposed in this pull request:**
Moved some resources of conferences_controller to new services to improve the code legebility
-
related issue: https://github.com/openSUSE/osem/issues/3036

### CODE REVIEW

File: app/controllers/conferences_controller.rb [hunk 1]
```diff
@@ -110,6 +46,51 @@ def calendar
 
   private
 
+  def handle_present_splash_page(splashpage)
+    @image_url = @service.conference_image_url(request)
+
+    if splashpage.include_cfp
+      cfp_variables
```
Comment (wenderjean) [inline app/controllers/conferences_controller.rb:53]: What do you think about naming this like "assign_cfp_variables"?

File: app/controllers/conferences_controller.rb [hunk 2]
```diff
@@ -110,6 +46,51 @@ def calendar
...
+    end
+
+    if splashpage.include_program
+      program_variables
```
Comment (wenderjean) [inline app/controllers/conferences_controller.rb:57]: Maybe "assign_program_variables"?

File: app/services/conferences_calendar_service.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+class ConferencesCalendarService
```
Comment (wenderjean) [inline app/services/conferences_calendar_service.rb:1]: What do you think about make this service stateless and delegate the responsibility of returning a full/not-full to itself? Suggestion:

```ruby
# Create or Build (not enough context)
class CreateConferenceCalendarService
  def self.call(calendar, conference, is_full_calendar)
    new(calendar, conference, is_full_calendar).call
  end
  
  def initialize(calendar, conference, is_full_calendar)
    @calendar = calendar
    @conference = conference
    @is_full_calendar = is_full_calendar
  end
  private_class_method :new
  
  def call
    if full?
      ...
    else
      ...
    end
  end
  
  private
  
  def full?
    @is_full_calendar
  end
end
```

**Comment by carlosdanielpohlod** [comment]
Sorry, I will open another PRs to apply these changes.

The calendar refactor is in this other one:
https://github.com/openSUSE/osem/pull/3101/

**Pull request closed without merging**